src/cargo/ops/registry.rs

```diff
@@ -35,11 +35,17 @@ pub struct PublishOpts<'cfg> {
     pub index: Option<String>,
     pub verify: bool,
     pub allow_dirty: bool,
+    pub dry_run: bool,
 }
 
 pub fn publish(ws: &Workspace, opts: &PublishOpts) -> CargoResult<()> {
     let pkg = try!(ws.current());
 
+    if !opts.verify && opts.dry_run {
+        bail!("cannot specify both no verify and dry run \
+               simultaneously");
+    }
```

Comment:
Could you clarify on what we're preventing with this check? I could imagine `--dry-run` also being done to check the `*.crate` manually without actually performing a build being a use case